### PR TITLE
Implemented parent-child inheritance for loggers

### PR DESCRIPTION
**Pull request by Niteshvgupta**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,6 +81,10 @@ loggers:
     myLogger:
         handlers: [console, info_file_handler]
         processors: [web_processor]
+    myLogger.mySubClass:
+        handlers: [console]
```
Comment (PhiloEpisteme) [inline README.md]: For clarity, lets use a different handler on the `mySubClass` logger.

File: examples/logger_config.yml [hunk 2]
```diff
@@ -34,6 +34,9 @@ handlers:
         stream: ./example_error.log
 
 loggers:
+    loggerA.child:
+        handlers: [console]
```
Comment (PhiloEpisteme) [inline examples/logger_config.yml]: Same as above. It'll be more clear to novice users if we use different handlers in the examples for child and parent loggers.

File: src/Config.php [hunk 3]
```diff
@@ -28,50 +28,57 @@ class Config
...
+     *
      * @var ConfigLoader
      */
     protected $loader = null;
 
     /**
      * Instantiate a Config object
      *
-     * @param string|array $input User input
+     * @param string|array $input  User input
```
Comment (PhiloEpisteme) [inline src/Config.php:81]: Is the two-space style here preferred for some reason?

File: src/Config.php [hunk 4]
```diff
@@ -165,13 +172,35 @@ protected function configureProcessors(array $processors)
...
      *
-     * @param  array $loggers Array of logger options
+     * @param array $loggers Array of logger options
      */
     protected function configureLoggers(array $loggers)
     {
+
+        $instantiatedLoggers = array();
+
+        if (array_key_exists('default', $loggers)) {
```
Comment (PhiloEpisteme) [inline src/Config.php:183]: I would suggest a comment above this to explain why you are doing the `default` logger first, and why you are removing it from the config file. It is fairly obvious to us, but a novice may appreciate the hint.

**Comment by PhiloEpisteme** [comment]
This looks excellent. The code is clean and easy to read and the style is fantastic. I left a few in-line comments for you. Some of them are simly things to discuss, others are suggestions and perhaps a nit or two.

I also think it would be useful to add 1 additional test case . . .
`$child_logger_name = 'foo.bar.baz';`
`$parent_logger_name = 'foo';`
We want to make sure that in that case, the parent gets put in place. Your code is correct it looks like, I just think this test case will help prevent us from accidentally screwing it up int he future.

Thanks for doing all of this work. I think we are basically done with this piece after this merge.

File: src/Config.php (hunk 3 above)
Comment (Niteshvgupta) [inline src/Config.php:81]: Yeah, according to `phpcs`, which Travis uses. It aligns the parameter name with the one just below it.

**Pull request merged**